server/src/test/java/org/elasticsearch/action/bulk/BulkRequestParserTests.java

```diff
@@ -30,7 +30,7 @@ public class BulkRequestParserTests extends ESTestCase {
 
     @UpdateForV10(owner = UpdateForV10.Owner.DATA_MANAGEMENT) // Replace with just RestApiVersion.values() when V8 no longer exists
     public static final List<RestApiVersion> REST_API_VERSIONS_POST_V8 = Stream.of(RestApiVersion.values())
-        .filter(v -> v.compareTo(RestApiVersion.V_8) > 0)
+        .filter(v -> v.major > RestApiVersion.V_8.major)
```

Comment:
For the Data Management folks,  Can you double check this?  Specifically, that values > V_8 is the desired behaviour?  

I think this was a bug. After removing V_7, this would return an empty list, causing tests to fail. This was previously returning a list containing only V_7 (as the ordinal for V_7 was greater than the ordinal for V_8). I don't think that behaviour was intended.

Reply:
@PeteGillinElastic can you confirm here?

Reply:
Yeah, this is my bad. Thanks for catching it. The intention was to include values later than `V_8` — but, of course, `compareTo` on enums compares by the order the values are defined in the enum, which in this case is backwards, so it was actually getting earlier than `V_8` (and even that behaviour was fragile as it depends on the declaration order which shouldn't be part of the contract).[*]

Your change looks good. Alternatively, I wonder whether `v.matches(RestApiVersion.onOrAfter(RestApiVersion.V_9))` might be even more explicit? Or `v != RestApiVersion.V_8`, which is equivalent once `V_7` goes away?

Aside: It feels like we should discourage the use of `compareTo` on `RestApiVersion`, since its behaviour is so counter-intuitive. I was going to suggest that we could override it with an implementation which just delegates to `super` and `@Deprecate` it — but it's declared as `final` on `Enum`, so that doesn't work. My personal opinion is that it was probably a mistake for `Enum` to implement `Comparable` in the first place, but we make use of it way to many times to consider banning it via static analysis.

[*] If, like me, you're wondering why the test passed with this bug: It's because in the code under test we effectively just test for `!= RestApiVersion.V_8`, which seems safe since we knew that `V_7` was going to go away before V9 shipped.

Reply:
Thanks @PeteGillinElastic , I updated this to use `v.matches(RestApiVersion.onOrAfter(RestApiVersion.V_9))` as per your suggestion.

Reply:
I don't see that change pushed here yet... But LGTM anyway.

Reply:
I'll leave @dakrone to give the overall approval for the DM stuff when he's ready.